File: cli/packages/cli/src/commands/list.ts
import { defineCommand } from "citty";
import { loadConfig, loadLock } from "@agpm/core";

export const listCommand = defineCommand({
  meta: {
    name: "list",
    description: "List configured artifacts and their status",
  },
  async run() {
    const cwd = process.cwd();
    const config = await loadConfig(cwd);
    const lock = await loadLock(cwd);

    if (config.artifacts.length === 0) {
      console.log("No artifacts configured.");
      console.log("\nAdd artifacts with: agpm add <source> <artifact>");
      return;
    }

    console.log("Configured artifacts:\n");

    for (const artifactRef of config.artifacts) {
      const lockEntry = lock.artifacts[artifactRef];

      console.log(`  ${artifactRef}`);

      if (lockEntry) {
        console.log(`    SHA: ${lockEntry.sha.slice(0, 8)}`);
        console.log(`    Path: ${lockEntry.path}`);
        if (lockEntry.metadata.description) {
          const desc =
            lockEntry.metadata.description.length > 60
              ? lockEntry.metadata.description.slice(0, 57) + "..."
              : lockEntry.metadata.description;
          console.log(`    ${desc}`);
        }
      } else {
        console.log(`    (not installed)`);
      }
      console.log();
    }

    console.log(`${config.artifacts.length} artifact(s) configured`);
  },
});

File: src/cli/list.rs
//! List command - list installed skills

use anyhow::Result;

use crate::config::{SkillsConfig, SkillsLock};

pub async fn run() -> Result<()> {
    let cwd = std::env::current_dir()?;
    
    let config = SkillsConfig::load(&cwd).await?;
    let lock = SkillsLock::load(&cwd).await?;
    
    if config.skills.is_empty() && config.marketplaces.is_empty() {
        println!("No skills configured.");
        println!("Run 'skills add <source>' to add skills.");
        return Ok(());
    }

    println!("Configured Skills:\n");
    
    // List individual skills
    if !config.skills.is_empty() {
        for skill in &config.skills {
            let locked = lock.skills.get(&skill.name);
            let status = if let Some(l) = locked {
                format!("@ {}", &l.sha[..8])
            } else {
                "not installed".to_string()
            };
            
            let desc = locked
                .and_then(|l| l.description.as_ref())
                .map(|d| format!(" - {}", d))
                .unwrap_or_default();
            
            println!("  {} ({}){}", skill.name, status, desc);
            println!("    source: {}", skill.source);
            if let Some(ref_) = &skill.ref_ {
                println!("    ref: {}", ref_);
            }
            if let Some(path) = &skill.path {
                println!("    path: {}", path);
            }
            println!();
        }
    }

    // List marketplace skills
    for marketplace in &config.marketplaces {
        if marketplace.enabled.is_empty() {
            continue;
        }
        
        println!("  Marketplace: {} ({})", marketplace.name, marketplace.source);
        for skill_name in &marketplace.enabled {
            let locked = lock.skills.get(skill_name);
            let status = if let Some(l) = locked {
                format!("@ {}", &l.sha[..8])
            } else {
                "not installed".to_string()
            };
            
            let desc = locked
                .and_then(|l| l.description.as_ref())
                .map(|d| format!(" - {}", d))
                .unwrap_or_default();
            
            println!("    - {} ({}){}", skill_name, status, desc);
        }
        println!();
    }

    Ok(())
}
